collector/lib/CollectorConfig.cpp

```diff
@@ -398,6 +399,9 @@ void CollectorConfig::HandleSinspEnvVars() {
 }
 
 bool CollectorConfig::YamlConfigToConfig(YAML::Node& yamlConfig) {
+  // Don't read the file during a scrape
+  std::lock_guard<std::mutex> lock(*mutex_);
+
   if (yamlConfig.IsNull() || !yamlConfig.IsDefined()) {
     CLOG(FATAL) << "Unable to read config from config file";
```

Comment:
This will cause collector to crash loop if the configuration is invalid, this is fine during startup, but not so good if collector was already running. Ideally, we would try to parse the configuration into a copy, check said configuration is valid and then swap it with the configuration being actively used. That way, if the configuration is changed in an invalid way, we could alert of this situation and keep collector running with the old configuration.

For the time being, we can keep the current implementation, but I would try to fix this ASAP.

Reply:
Now this only fatal is `runtime_config_` is not set.